web-api/src/mind.rs

```diff
@@ -231,6 +315,124 @@ where
         .map_err(Into::into)
 }
 
+// function that assigns a score to a vector of documents based on the user's interests
+// score is equal to 1 if the document is interesting to the user, 0 otherwise
+fn score_documents(documents: &[&Document], user_interests: &[String]) -> Array1<f32> {
+    let scores = documents
+        .iter()
+        .map(|document| {
+            if document.is_interesting(user_interests) {
+                2.0
+            } else {
+                0.0
+            }
+        })
+        .collect::<Vec<f32>>();
+    Array1::from(scores)
+}
+
+fn write_array<T, S, D>(writer: impl io::Write, array: &ndarray::ArrayBase<S, D>) -> io::Result<()>
+where
+    T: Clone + npyz::AutoSerialize,
+    S: ndarray::Data<Elem = T>,
+    D: ndarray::Dimension,
+{
+    let shape = array.shape().iter().map(|&x| x as u64).collect::<Vec<_>>();
+    let c_order_items = array.iter();
+
+    let mut writer = npyz::WriteOptions::new()
+        .default_dtype()
+        .shape(&shape)
+        .writer(writer)
+        .begin_nd()?;
+    writer.extend(c_order_items)?;
+    writer.finish()
+}
+
+/// Runs persona based benchmark
+async fn run_persona_based_benchmark() -> Result<(), Error> {
+    let document_provider = DocumentProvider::new("news.tsv")?;
+    let state = State::new(Storage::default()).unwrap();
+    // load documents from document provider to state
+    state
+        .insert(
+            document_provider
+                .documents
+                .values()
+                .cloned()
+                .collect::<Vec<_>>(),
+        )
+        .await
+        .unwrap();
+    let benchmark_config = PersonaBasedConfig::default();
+    // create 3d array of zeros with shape (nranks, iterations, n_documents)
+    let mut results = Array3::<f32>::zeros(Dim([
+        benchmark_config.nranks.len(),
+        benchmark_config.iterations,
+        benchmark_config.n_documents,
+    ]));
```

Comment:
:art: the array constructors accepts any kind of shape-like object, we could just pass an array and avoid the extra `Dim`
```suggestion
    let mut results = Array3::<f32>::zeros([
        benchmark_config.nranks.len(),
        benchmark_config.iterations,
        benchmark_config.n_documents,
    ]);
```

Reply:
I don't know why it was not working previously, but now when I tried this it worked